### ISSUE TITLE
Add ArcFace loss.

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#546]
**System information**.

TensorFlow version (you are using): 2.8
Are you willing to contribute it (Yes/No) : No (but willing to provide starter if required.)

**Describe the feature and the current behavior/state**.

**ArcFace**, or Additive Angular Margin Loss, is a loss function used mainly in face recognition tasks. It is used to obtain highly discriminative features. This loss function can be used in <del>both</del> vision task <del>and nlp task</del>. <del>So, I think its general for `keras_cv` and `keras_nlp`</del>.

Paper: https://arxiv.org/pdf/1801.07698.pdf,
TL,DR, https://paperswithcode.com/method/arcface
Citation: 2939  (current)

**Will this change the current api? How?**

```python

tf.keras.losses.SparseArcFace
tf.keras.losses.CategoricalArcFace
```

**Who will benefit from this feature?**

`keras` practitioners. 

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**
- Briefly describe your candidate solution(if contributing):
     - https://github.com/4uiiurz1/keras-arcface

**Comment by ayulockin** [comment]
Hey, @LukeWood @sayakpaul I think this would be a great feature addition to KerasCV. It's especially popular in Kaggle competitions for reference. 

Please assign it to me.

**Comment by sayakpaul** [comment]
I concur. Especially for retrieval tasks, ArcFace loss is greatly beneficial.

**Comment by LukeWood** [comment]
interesting, can this be used with the existing models we have?  i.e. if we add this will we have the tools we need to support facial recognition?

**Comment by sayakpaul** [comment]
Should be compatible with existing models.

**Comment by LukeWood** [comment]
So theoretically, we include this, and use this loss to create a facial recognition tutorial with the other existing components?

**Comment by sayakpaul** [comment]
Shoul be the case. We have a tutorial on few-shot learning and we coul definitely take some components from that too.

**Comment by LukeWood** [comment]
> Shoul be the case. We have a tutorial on few-shot learning and we coul definitely take some components from that too.

ok, sounds good.  Then I think this is a great component.  Especially if we can include a tutorial on facial recognition.

**Comment by sayakpaul** [comment]
@ayulockin take it for the grab if you're up.

**Comment by ayulockin** [comment]
This loss is compatible with existing models. 

I can take this up. Will start working on it soon. Also the idea of tutorial is great for this loss.

**Comment by yashchks87** [comment]
Would love to work on this if no one has picked up still.

**Comment by alexandrekj** [comment]
Any updates on this?

**Comment by jbischof** [comment]
It looks like this fell through the cracks @alexandrekj. @yashchks87 or @alexandrekj, are either of you interested in working on this?

**Comment by yashchks87** [comment]
I can take on it. @jbischof

**Comment by DLumi** [comment]
Why this was closed, though? 
I cannot see any implementations of this loss anywhere in the package.

@innat @yashchks87 @jbischof
